test/lms_test.c

```diff
@@ -318,6 +317,37 @@ static int lms_signing_fail_test(void)
     return ret;
 }
 
+static int lms_message_signing_fail_test(void)
+{
+    int ret = 0;
+    LMS_ACVP_TEST_DATA *td = &lms_testdata[0];
+    EVP_PKEY_CTX *ctx = NULL;
+    EVP_PKEY *pkey = NULL;
+    EVP_SIGNATURE *sig = NULL;
+
+    ret = TEST_ptr(pkey = lms_pubkey_from_data(td->pub, td->publen))
+        && TEST_ptr(sig = EVP_SIGNATURE_fetch(libctx, "LMS", NULL))
+        && TEST_ptr(ctx = EVP_PKEY_CTX_new_from_pkey(libctx, pkey, NULL))
+        && TEST_int_eq(EVP_PKEY_sign_message_init(ctx, sig, NULL), -2);
+
+    EVP_PKEY_free(pkey);
+    EVP_PKEY_CTX_free(ctx);
+    EVP_SIGNATURE_free(sig);
+    return ret;
+}
+
+static int lms_keygen_fail_test(void)
+{
+    int ret;
+    EVP_PKEY_CTX *ctx = NULL;
+
+    ret = TEST_ptr(ctx = EVP_PKEY_CTX_new_from_name(libctx, "LMS", NULL))
+        && TEST_int_eq(EVP_PKEY_paramgen_init(ctx), -2);
```

Comment:
Slightly strange that the test is called keygen_fail, but we actually cal paramgen_init. But ok.